### PR TITLE
Add deferred Payments for In Page Checkout

### PR DESCRIPTION
**Pull request by ClaireAlma**
### Reason for change

[Linear task](https://linear.app/almapay/issue/ECOM-1503/[%F0%9F%A7%A9-ps]-fix-sonarcloud-issues)

### How to test

- mount an AC infra
- activate in page checkout
- we will see deferred payments in page
- launch unit tests in ssh

### CODE REVIEW

File: Helpers/PaymentPlansHelper.php [hunk 1]
```diff
@@ -101,7 +101,8 @@ public function paymentTriggerIsAllowed(): bool
      */
     public function isInPageAllowed(string $planKey): bool
     {
-        $planIsAllowed = preg_match('!general:[1-4]:0:0!', $planKey);
+        $planIsAllowed = preg_match('!general:[1-4]:([\d]+):([\d]+)!', $planKey);
```
Comment (Francois-Gomis) [inline Helpers/PaymentPlansHelper.php:104]: Deferred month is allow ?

File: Test/Unit/Helpers/PaymentPlansHelperTest.php [hunk 2]
```diff
@@ -112,6 +112,50 @@ public function testFormatFeePlanConfigForBackOfficeDisplay($apiFeePlan, $config
...
+            $this->assertEquals($plan['result'], $this->createPaymentPlansHelper()->isInPageAllowed($plan['key']));
+        }
+    }
+
+    /**
+     * @return void
+     */
+    public function testIsInPageAllowedWithoutInPageActivated()
+    {
+        $planKeys = $this->getInPagePlanKeys();
```
Comment (Francois-Gomis) [inline Test/Unit/Helpers/PaymentPlansHelperTest.php:133]: You can create a dataProvider for testing separatly each plan

File: view/frontend/web/js/view/payment/method-renderer/alma_monthly_payments.js [hunk 3]
```diff
@@ -228,16 +228,21 @@ define(
                 if(!plan.inPageAllowed) {
                     return true;
                 }
+console.log('alma-in-page'+'-'+self.checkedPaymentMethod())
```
Comment (Francois-Gomis) [inline view/frontend/web/js/view/payment/method-renderer/alma_monthly_payments.js:231]: log in prod

File: Test/Unit/Helpers/PaymentPlansHelperTest.php [hunk 4]
```diff
@@ -112,6 +112,50 @@ public function testFormatFeePlanConfigForBackOfficeDisplay($apiFeePlan, $config
         $this->assertEquals($result, $this->createPaymentPlansHelper()->formatLocalFeePlanConfig($apiFeePlan, $configFeePlan));
     }
 
+    /**
+     * @return void
+     */
+    public function testIsInPageAllowedWithInPageActivated()
+    {
+        $planKeys = $this->getInPagePlanKeys();
```
Comment (Francois-Gomis) [inline Test/Unit/Helpers/PaymentPlansHelperTest.php:120]: You can create a dataProvider for testing separatly each plan

File: Helpers/PaymentPlansHelper.php (hunk 1 above)
Comment (ClaireAlma) [reply Helpers/PaymentPlansHelper.php:104]: Yes all deferred payments

File: Test/Unit/Helpers/PaymentPlansHelperTest.php (hunk 2 above)
Comment (Benjamin-Freoua-Alma) [reply Test/Unit/Helpers/PaymentPlansHelperTest.php]: @Francois-Gomis 
Done

File: Test/Unit/Helpers/PaymentPlansHelperTest.php [hunk 5]
```diff
@@ -112,6 +112,50 @@ public function testFormatFeePlanConfigForBackOfficeDisplay($apiFeePlan, $config
...
+        foreach ($planKeys as $plan) {
+            $this->configHelper->method('isInPageEnabled')->willReturn(true);
+            $this->assertEquals($plan['result'], $this->createPaymentPlansHelper()->isInPageAllowed($plan['key']));
+        }
+    }
+
+    /**
+     * @return void
+     */
+    public function testIsInPageAllowedWithoutInPageActivated()
```
[the remark this answers is not in the document]
Comment (Benjamin-Freoua-Alma) [reply Test/Unit/Helpers/PaymentPlansHelperTest.php]: @Francois-Gomis Done

File: Test/Unit/Helpers/PaymentPlansHelperTest.php [hunk 6]
```diff
@@ -112,6 +112,50 @@ public function testFormatFeePlanConfigForBackOfficeDisplay($apiFeePlan, $config
         $this->assertEquals($result, $this->createPaymentPlansHelper()->formatLocalFeePlanConfig($apiFeePlan, $configFeePlan));
     }
 
+    /**
+     * @return void
+     */
+    public function testIsInPageAllowedWithInPageActivated()
```
[the remark this answers is not in the document]
Comment (Benjamin-Freoua-Alma) [reply Test/Unit/Helpers/PaymentPlansHelperTest.php]: @Francois-Gomis Done

File: Test/Unit/Helpers/PaymentPlansHelperTest.php (hunk 4 above)
Comment (Benjamin-Freoua-Alma) [reply Test/Unit/Helpers/PaymentPlansHelperTest.php]: @Francois-Gomis Done

Comment (Francois-Gomis) [reply Test/Unit/Helpers/PaymentPlansHelperTest.php]: assertFalse c'est nickel je l'avais pas vu

File: view/frontend/web/js/view/payment/method-renderer/alma_monthly_payments.js (hunk 3 above)
Comment (Benjamin-Freoua-Alma) [reply view/frontend/web/js/view/payment/method-renderer/alma_monthly_payments.js]: @Francois-Gomis Done

**Pull request merged**